pkgs/tools/package-management/nix-serve/default.nix

```diff
@@ -12,23 +21,34 @@ stdenv.mkDerivation rec {
     sha256 = "0mjzsiknln3isdri9004wwjjjpak5fj8ncizyncf5jv7g4m4q1pj";
   };
 
-  buildInputs = [ perl nix ]
-    ++ (with perlPackages; [ DBI DBDSQLite Plack Starman ]);
+  buildInputs = [ makeWrapper ];
 
-  dontBuild = false;
+  propagatedBuildInputs = [
+    nix
+    perl
+    perlPackages.DBDSQLite
+    perlPackages.DBI
+    perlPackages.Plack
+    perlPackages.Starman
+  ];
+
+  phases = ["unpackPhase" "installPhase"];
```

Comment:
It might not make any difference in this case, but in general, using explicit phases is a bad idea, because you might miss out on useful phases you didn't know about or that were added after the fact. Disabling problematic phases is almost always the correct solution.

For a good example of the perils of explicit phases, see 85b87796ae2506b7d77040ac873df17580bffce7. Here, the use of explicit phases had caused gcc and friends to be included in the closure, greatly increasing its size.